src/SourceLocator/SourceStubber/PhpStormStubsSourceStubber.php

```diff
@@ -415,6 +445,57 @@ private function getAbsoluteFilePath(string $filePath): string
         return sprintf('%s/%s', $this->getStubsDirectory(), $filePath);
     }
 
+    /**
+     * @param list<Node\Name> $nameNodes
+     *
+     * @return list<Node\Name>
+     */
+    private function modifyExtendsImplementsByPhpVersion(string $className, array $nameNodes): array
+    {
+        $modifiedNames = [];
+        foreach ($nameNodes as $nameNode) {
+            $name = $nameNode->toString();
+
+            if ($className === ParseError::class) {
+                if ($name === CompileError::class && $this->phpVersion < 70300) {
+                    $modifiedNames[] = new Node\Name\FullyQualified(Error::class);
+                    continue;
+                }
+            } elseif ($className === SplFixedArray::class) {
+                if ($name === JsonSerializable::class && $this->phpVersion < 80100) {
+                    continue;
+                }
+
+                if ($name === IteratorAggregate::class && $this->phpVersion < 80000) {
+                    continue;
+                }
+
+                if ($name === Iterator::class && $this->phpVersion >= 80000) {
+                    continue;
+                }
+            } elseif ($className === SimpleXMLElement::class) {
+                if ($name === RecursiveIterator::class && $this->phpVersion < 80000) {
+                    continue;
+                }
+            } elseif (in_array($className, [DatePeriod::class, PDOStatement::class, true])) {
+                if ($name === IteratorAggregate::class && $this->phpVersion < 80000) {
+                    $modifiedNames[] = new Node\Name\FullyQualified(Traversable::class);
+                    continue;
+                }
+            }
+
+            $nodeData = $this->getClassNodeData($name);
+
+            if ($nodeData === null) {
+                continue;
+            }
+
+            $modifiedNames[] = $nameNode;
```

Comment:
If I understand it correctly, except for `ParseError`, `CompileError`, `SplFixedArray`, `SimpleXMLElement`, `DatePeriod` and `PDOStatement`, `$modifiedNames === $namedNode`?

Can we somehow short-circuit that perhaps? Doesn't need to be, if the code gets even worse, but would be best to avoid

Reply:
No, there are some others that are fixed automatically by `if ($this->getClassNodeData($name) === null) {`, eg. `DOMDocument` implements `DOMParentNode` that is PHP 8+ only.